torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -244,13 +247,84 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                         )
 
                     node.replace_all_uses_with(replacement_proxy.node)
-
                     graph.erase_node(node)
 
-    # DCE symbols (which are guaranteed to be pure) only
-    for proxy in reversed(expr_to_sym_proxy.values()):
-        if len(proxy.node.users) == 0 and proxy.node.op != "placeholder":
-            graph.erase_node(proxy.node)
+    # Now do one more pass that specializes all symfloats we didn't manage
+    # to tensorify away.
+    nodes = list(graph.nodes)
+    for i, node in reversed(list(enumerate(nodes))):
+        if node.op == "output" or node.op == "placeholder":
+            continue
+
+        with graph.inserting_before(nodes[i]):
+            if len(node.users) == 0 and not node.is_impure():
+                graph.erase_node(node)
+                continue
+
+            args = []
+            kwargs: Dict[str, Any] = {}
+            transform = False
+            for a in node.args:
+                if (
+                    isinstance(a, fx.Node)
+                    and (zf := a.meta.get("val")) is not None
+                    and isinstance(zf, (torch.SymFloat, torch.SymInt))
+                    and (node_hint := zf.node.hint) is not None
+                    # The high level idea here is to specialize if we have
+                    # a node that has exclusively SymFloat free symbols.
+                    # This is because we don't want to specialize cases like
+                    # math.floor(math.sqrt(x.size(0))).
```

Comment:
Here is a near miss. Suppose we have `some_op(math.sqrt(x.size(0)) + input_float)`.  Here, there is a dependence on both `s0` and on `zf0`. Let's say we were not able to tensorify `some_op`. We would like to force `input_float` to be specialized. However, because there is an `s0` free symbol here, we don't guard on this expression (and rightly so.)

Will `input_float` eventually get specialized later in the pass? It's not clear that this happens. The problem is that the pass here only inspects args/kwargs of operators it's trying to specialize. So if `input_float` is literally the input argument, you will never force spec, which seems bad.